src/lib/Components/Bidding/Screens/ConfirmBid/PriceSummary.tsx

```diff
@@ -0,0 +1,95 @@
+import { Box, Flex, Sans, Serif } from "@artsy/palette"
+import React from "react"
+import { createFragmentContainer, graphql, QueryRenderer } from "react-relay"
+
+import { Bid } from "lib/Components/Bidding/types"
+import { defaultEnvironment } from "lib/relay/createEnvironment"
+import renderWithLoadProgress from "lib/utils/renderWithLoadProgress"
+
+import { PriceSummary_calculatedCost } from "__generated__/PriceSummary_calculatedCost.graphql"
+import { PriceSummaryQuery, PriceSummaryQueryVariables } from "__generated__/PriceSummaryQuery.graphql"
+
+interface PriceSummaryViewProps {
+  calculatedCost: PriceSummary_calculatedCost
+  bid: Bid
+}
+
+const _PriceSummary = ({ bid, calculatedCost }: PriceSummaryViewProps) => (
+  <Box mt={4} mx={4}>
+    <Serif mb={1} size="4" weight="semibold" color="black100">
+      Summary
+    </Serif>
+
+    <Flex mb={1} flexDirection="row" justifyContent="space-between">
+      <Sans size="3" color="black100">
+        Your max bid
+      </Sans>
+      <Sans size="3" color="black100">
+        {`${bid.display}.00`}
+      </Sans>
+    </Flex>
+
+    <Flex mb={1} flexDirection="row" justifyContent="space-between">
+      <Sans size="3" color="black100">
+        Buyer’s premium
+      </Sans>
+      <Sans size="3" color="black100">
+        {calculatedCost.buyersPremium.display}
+      </Sans>
+    </Flex>
+
+    <Flex mb={1} flexDirection="row" justifyContent="space-between">
+      <Sans size="3" color="black100">
+        Subtotal
+      </Sans>
+      <Sans size="3" color="black100">
+        {calculatedCost.subtotal.display}
+      </Sans>
+    </Flex>
+
+    <Sans size="3" color="black60">
+      Plus any applicable shipping, taxes, and fees.
+    </Sans>
+  </Box>
+)
+
+const PriceSummaryFragmentContainer = createFragmentContainer(_PriceSummary, {
+  calculatedCost: graphql`
+    fragment PriceSummary_calculatedCost on CalculatedCost {
+      buyersPremium {
+        display
+      }
+      subtotal {
+        display
+      }
+    }
+  `,
+})
+
+interface PriceSummaryProps extends Partial<PriceSummaryQueryVariables> {
```

Comment:
why is this a `Partial<T>` type? I guess considering that we know that the UI is providing a bid object, what protection/value does making this a union type of the query variables + others provide? Perhaps we can talk through this for a min so that I can learn here.

Reply:
The `PriceSummaryQueryVariables` interface is auto-generated by Relay so it'll add a little safety for when we update the query argument. It has two arguments: `saleArtworkId` and `bidAmountCents`. I made it a partial since the object that gets injected into this component is a `Bid` object (we use both `bid.cents` and `bid.display` to render the component), and we don't use the other `bidAmountCents` argument in the `PriceSummaryQueryVariables` interface.